### PR TITLE
#37 custom error message handling

### PR DESCRIPTION
**Pull request by wanne27**

### CODE REVIEW

File: Recrutify/Recrutify.Host/Startup.cs [hunk 1]
```diff
@@ -127,6 +128,10 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             {
                 app.UseDeveloperExceptionPage();
             }
+            else
+            {
+                app.UseMiddleware<ExceptionMiddleware>();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Startup.cs:133]: wtite extension method like in example

File: Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        }
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            try
+            {
+                await _next(httpContext);
+            }
+            catch
+            {
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs:24]: log error like in example

File: Recrutify/Recrutify.Host/Startup.cs (hunk 1 above)
Comment (wanne27) [reply Recrutify/Recrutify.Host/Startup.cs:132]: should i create a static method?

File: Recrutify/Recrutify.Host/CustomExceptionMiddleware/ConfigureExceptionMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Recrutify.Host.Exceptions;
+
+namespace Recrutify.Host.CustomExceptionMiddleware
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/CustomExceptionMiddleware/ConfigureExceptionMiddleware.cs:4]: create folder Extensions and move it here

File: Recrutify/Recrutify.Host/CustomExceptionMiddleware/ConfigureExceptionMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public static class ConfigureExceptionMiddleware
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/CustomExceptionMiddleware/ConfigureExceptionMiddleware.cs:6]: name MiddlewareExtensions

File: Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Recrutify.Host.Exceptions
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs:6]: move to Infrastructure folder

File: Recrutify/Recrutify.Host/Extensions/MiddlewareExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using Microsoft.AspNetCore.Builder;
+
+namespace Recrutify.Host.Exceptions
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Extensions/MiddlewareExtensions.cs:3]: update namespace name

Comment (wanne27) [reply Recrutify/Recrutify.Host/Extensions/MiddlewareExtensions.cs:3]: this is right? namespace Recrutify.Host.Exceptions

Comment (elizavetapavliv) [reply Recrutify/Recrutify.Host/Extensions/MiddlewareExtensions.cs:3]: but your folder named Extensions, folder and namespace should be same

File: Recrutify/Recrutify.Host/Extensions/MiddlewareExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Recrutify.Host.CustomExceptionMiddleware;
+
+namespace Recrutify.Host.Extensions
+{
+    public static class MiddlewareExtensions
+    {
+        public static void ConfigureCustomExceptionMiddleware(this IApplicationBuilder app)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Extensions/MiddlewareExtensions.cs:8]: rename UseHttpStatusExceptionHandler

File: Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs [hunk 8]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Recrutify.Host.CustomExceptionMiddleware
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs:6]: move to Infrastructure folder / namespace

Comment (wanne27) [reply Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs:6]: I dont understand

Comment (elizavetapavliv) [reply Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs:6]: ExceptionMiddleware should be located in Infrastructure folder of Recrutidy.Host project. Now it is located in CustomExceptionMiddleware folder - it's incorrect. Namespace should be changed too from Recrutify.Host.CustomExceptionMiddleware to Recrutify.Host.Infrastructure

Comment (wanne27) [reply Recrutify/Recrutify.Host/CustomExceptionMiddleware/ExceptionMiddleware.cs:6]: ok

File: Recrutify/Recrutify.Host/Infrastructure/ExceptionMiddleware.cs [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            try
+            {
+                await _next(httpContext);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Something went wrong: {ex}");
+                httpContext.Response.StatusCode = 500;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Infrastructure/ExceptionMiddleware.cs:28]: do not hardcode 500 : use Microsoft static class StatusCodes

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**